release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+import static java.lang.String.format;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * ReleaseNoteCreation creates the body of the release note for a repository of Scalar products. The
+ * body of the release note is written out as a markdown to standard output.
+ *
+ * <p>This script is assumed to be executed in a GitHub Actions workflow.
+ *
+ * <p>Note that it is needed that java 11 to execute this script since it is executed as a
+ * Single-File Source-code program.
+ */
+@SuppressWarnings("DefaultPackage")
+public class ReleaseNoteCreation {
+
+  private static final String DEBUG = System.getenv("DEBUG");
+
+  private final GitHubContext ghContext;
+
+  final Map<Category, List<ReleaseNoteText>> categoryMap = new EnumMap<>(Category.class);
+  final Map<String, List<ReleaseNoteText>> sameAsItems = new HashMap<>();
+
+  public static void main(String... args) throws Exception {
+    if (args.length != 4) {
+      System.err.printf(
+          "Usage:%n    java --source 11 %s.java <owner> <projectTitlePrefix> <version>"
+              + " <repository>%n%nExample:%n    java --source 11 %s.java scalar-labs ScalarDB 4.0.0"
+              + " scalardb%n",
+          ReleaseNoteCreation.class.getSimpleName(), ReleaseNoteCreation.class.getSimpleName());
+      System.exit(1);
+    }
+
+    String owner = args[0];
+    String projectTitlePrefix = args[1];
+    String version = args[2];
+    String repository = args[3];
```

Comment:
Don't we need to validate these passed arguments here?

Reply:
The script assumed to run on the GitHub Actions workflow and the parameters are passed in the workflow automatically. Thus I didn't write validation for the arguments.

What do you think? Even so, do you think the script needs to validate the argument?

Reply:
Okay. That's reasonable. It would be better to leave the code comment describing that those arguments were passed from the GitHub Actions.

Reply:
Thank you for the confirmation. Done at 2c239a5